### PR TITLE
RSS Enhancements

### PR DESCRIPTION
**Pull request by aormsby**
As discussed in #147, I have added these two options (copied from README update):

**A page's cover image now appears at the top of its feed display**. This image is set manually using the cover params. If unset, the RSS generator searches for the first image file in the page bundle whose name includes 'featured', 'cover', or 'thumbnail'.

**You can optionally display the full page content in your RSS feed** (default is Description or Summary data from Front Matter). Set `rssFullText = true` in your `config.toml` file to enable this option.

**RSS validated and ready to go!**

### CODE REVIEW

**Comment by aormsby** [comment]
This is good to go, by the way. No more last-minute fixes from me.

**Comment by aormsby** [comment]
Oh wait, I did a dumb thing. I typed `image` instead of `img` for tags. Fixing momentarily.

File: README.md [hunk 1]
```diff
@@ -10,24 +10,22 @@ This theme is pretty basic and covers all of the essentials. All you have to do
...
+- [How to add a cover image to your posts](#how-to-add-a-cover-image-to-your-posts)
+- [How to display the Last Modified Date in your posts](#how-to-display-the-last-modified-date-in-your-posts)
+- [How to hide "Read more" button](#how-to-hide-read-more-button)
+- [Add-ons](#add-ons)
+- [How to run your site](#how-to-run-your-site)
+- [How to edit the theme](#how-to-edit-the-theme)
+- [How to contribute](#how-to-contribute)
+- [`Hello Friend` theme user?](#hello-friend-theme-user)
+- [Sponsoring](#sponsoring)
+- [License](#license)
```
Comment (aormsby) [inline README.md]: It looks like we have different settings for TOC generation. I guess I'm ignoring H1s. Do you want me to add those back in, or is this fine?

File: layouts/_default/rss.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    {{ range $pages }}
+    <item>
+      <title>{{ .Title }}</title>
+      <link>{{ .Permalink }}</link>
+      <pubDate>{{ .Date.Format "Mon, 02 Jan 2006 15:04:05 -0700" | safeHTML }}</pubDate>
+      {{ with .Site.Author.email }}<author>{{.}}{{ with $.Site.Author.name }} ({{.}}){{end}}</author>{{end}}
+      <guid>{{ .Permalink }}</guid>
+      <description>
+        {{ if .Params.Cover }}
+          {{ if .Params.UseRelativeCover }}{{ printf "%s%s%s%s" "<img src=\"" .Permalink .Params.Cover "\"/>"}}
```
Comment (panr) [inline layouts/_default/rss.xml:45]: Huh, I didn't know that `printf` works with syntax like this 🤔

Comment (aormsby) [reply layouts/_default/rss.xml:45]: I did have to play around with quote placement until I got functional results, but I made sure to check the output repeatedly. I always seem to have trouble with string formatting, though. I'm certainly open to ideas here.

File: layouts/_default/rss.xml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    {{ end }}
+    {{ range $pages }}
+    <item>
+      <title>{{ .Title }}</title>
+      <link>{{ .Permalink }}</link>
+      <pubDate>{{ .Date.Format "Mon, 02 Jan 2006 15:04:05 -0700" | safeHTML }}</pubDate>
+      {{ with .Site.Author.email }}<author>{{.}}{{ with $.Site.Author.name }} ({{.}}){{end}}</author>{{end}}
+      <guid>{{ .Permalink }}</guid>
+      <description>
+        {{ if .Params.Cover }}
```
[the remark this answers is not in the document]
Comment (aormsby) [reply layouts/_default/rss.xml:44]: I use Feedly, too! :) I think there's a lot to unwrap here about images in feeds and Feedly. I've had to rewrite this twice just to try and keep things simple, so hopefully this all makes sense.

### XML <image> tag for RSS
I had to read up on W3 for this stuff as well, and that `<image>` tag is for the RSS channel as a whole. It does not display images inside the content (which is all HTML format held inside the `<description>` tag). I did add an XML `<image>` tag near the top of the RSS generator, but I'm unsure if it was successful. (see Feedly update issues)

### Feedly cover images in list view (Part 1 - why they're still missing)
Feedly will not update existing feed content, even if the RSS data details change. All old post data will be displayed the same forever. If it had no cover image before, it won't update to include one. At least, with all my testing I haven't found a single way to update my feed, and it's really annoying. Only newly published items seem to have their data read fresh. I think I will create a test site with an RSS feed that we can publish stuff to and see our various code updates.

### Feedly cover images in list view (Part 2 - they aren't RSS-based)
The post image in Feedly's list view doesn't actually come from the RSS data. (At least, I'm 95% sure of this.) It comes from the Hugo-generated [Open Graph data here](https://github.com/gohugoio/hugo/blob/25a6b33693992e8c6d9c35bc1e781ce3e2bca4be/tpl/tplimpl/embedded/templates/opengraph.html#L5-L16). So if that data isn't present in your generated pages because your image data hasn't been found by Hugo, the image won't show up in the list view

Screen Shot 2020-05-26 at 14 28 02
(Coincidentally, this is also the incorrect cover image, because it wouldn't update for me.)

The cover image data that I added into the RSS feed (which is just an HTML `` inside the XML `<description>` tag) appears at the top of the actual content view like this.

Screen Shot 2020-05-26 at 14 33 49

I'm a little worried that other feed readers will not display things the same way (or possibly at all), but I'm at least sure this is what should occur in Feedly.

Check out my RSS feed for adamormsby.com. If you haven't looked at my feed in Feedly yet, you may actually get the most updated version of my site's output. Perhaps it will show you what I mean. (Also, I'd love to see it if you do!)

**TL;DR:** I think all my stuff is correct, but it's worth testing again on a new test site.

Do these explanations help?

[the remark this answers is not in the document]
Comment (aormsby) [reply layouts/_default/rss.xml:44]: Also, I added you to a small testbed. Feel free to run any tests and push any new posts to see if everything works as expected. Hopefully, we can sort out what's working, what isn't, and what's Feedly's fault. 😆

[the remark this answers is not in the document]
Comment (panr) [reply layouts/_default/rss.xml:44]: Whoa, great explanation 👍 I've just tested it again with my other Hugo based sites and you're 100% right. `Image` is not RSS based and it's a little fucked up in Feedly 😅

My only concern is if we should keep relative paths here https://github.com/panr/hugo-theme-hello-friend/pull/148/files#diff-cd5b9b2f01fe87b55aab7340a230e838R45 and here  https://github.com/panr/hugo-theme-hello-friend/pull/148/files#diff-cd5b9b2f01fe87b55aab7340a230e838R51 🤔 I spotted this issue in some random free RSS viewers like https://feed.mikle.com, where the paths incorrectly direct to the `RSS viewer domain` + `/relative path to the image`

File: layouts/_default/rss.xml (hunk 2 above)
Comment (panr) [reply layouts/_default/rss.xml:45]: I use it like:

```
{{ printf "<img src=\" %s%s\"/>"  .Permalink .Params.Cover }}
```

**Comment by panr** [comment]
⚠️ NOTE TO SELF: Remember to close #147.

File: layouts/_default/rss.xml (hunk 2 above)
Comment (aormsby) [reply layouts/_default/rss.xml:45]: Wow, I like that a **lot** more. 😆 I just never ran across any examples like that. I'll definitely update that.

Comment (aormsby) [reply layouts/_default/rss.xml:45]: I'm going to resolve this comment section with the hope that Github's UI will clear up a little. I'm having a hard time seeing the diffs you linked elsewhere.

File: layouts/_default/rss.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (aormsby) [reply layouts/_default/rss.xml:44]: Hmm... I'm not sure I understand. How is the RSS domain getting into the URL at all? Hugo builds everything locally into the `public/index.xml` file, and I'm seeing correct full domain links in mine. Here's some of my output for that section:

`&lt;img src=&#34;https://www.adamormsby.com/posts/hugo-themes-making-modifications/cover.jpg&#34;/&gt;`
(The escaped characters convert later, of course.)

Could it be that the RSS viewer is modifying them when it pulls the site's RSS data?

[the remark this answers is not in the document]
Comment (panr) [reply layouts/_default/rss.xml:44]: Hmm... maybe it's something with Vercel, but check deployed test site and https://hello-friend.now.sh/index.xml. The URLs for the images (I'm using ``) are somehow relative 🤔 I think something is messed up with `baseURL` on the Vercel side ¯\_(ツ)_/¯

I see that your site has proper URLs... So we can leave it as it is for now 😉

**Comment by panr** [comment]
@aormsby I'm ready to merge it. Let me know if it contains everything you wanted to add 😉

File: layouts/_default/rss.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (aormsby) [reply layouts/_default/rss.xml:44]: Hmm yeah, could be. I've certainly shot myself in the foot before by forgetting to update my baseURL. Those kinds of issues can hide for days!

**Comment by aormsby** [comment]
I think we're good! I never did have luck testing the overall site image, but it doesn't cause any problems. I'll just make another PR if I ever figure that out. Cheers, mate!

**Comment by panr** [comment]
Merged! Thanks a lot for this! Cheers! 🍷
